overlord/snapstate/backend/setup.go

```diff
@@ -30,15 +30,27 @@ import (
 )
 
 // SetupSnap does prepare and mount the snap for further processing.
-func (b Backend) SetupSnap(snapFilePath string, sideInfo *snap.SideInfo, meter progress.Meter) error {
+func (b Backend) SetupSnap(snapFilePath string, sideInfo *snap.SideInfo, meter progress.Meter) (err error) {
 	// This assumes that the snap was already verified or --dangerous was used.
 
-	s, snapf, err := OpenSnapFile(snapFilePath, sideInfo)
+	var s *snap.Info
+	var snapf snap.Container
+	s, snapf, err = OpenSnapFile(snapFilePath, sideInfo)
```

Comment:
Nitpick: why explicit var declarations? I think the compact version looked nicer.

Reply:
Couldn't use `s, snapf, err := OpenSnapFile(snapFilePath, sideInfo)` as that would shadow `err`.

Reply:
Would it work if you drop the `(err error)` from the return value and just use `error`

Reply:
No, AFAICT named arguments is the only way of getting access to returned values in deferred funcs. Let me try to rename that particular err in OpenSnapFile to something else, eg:
```
s, snapf, openErr := OpenSnapFile(snapFilePath, sideInfo)
if openErr != nil {
    return openErr
}
```
Does this look ok?

Reply:
Yeah, that is fine.